nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/LogManager.java

```diff
@@ -163,19 +166,42 @@ default LogLag getLag(String name, String group) {
      * <br/>
      * Two functions need to be provided to extract the timestamp and a key from a record.
      *
-     * @since 10.1
+     * @since 10.2
      */
-    <M extends Externalizable> List<Latency> getLatencyPerPartition(String name, String group,
+    <M extends Externalizable> List<Latency> getLatencyPerPartition(String name, String group, Codec<M> codec,
             Function<M, Long> timestampExtractor, Function<M, String> keyExtractor);
 
+    /**
+     * Returns the lag with latency. Timestamps used to compute the latencies are extracted from the records. This
+     * requires to read one record per partition so it costs more than {@link #getLagPerPartition(String, String)}.
+     * <br/>
+     * Two functions need to be provided to extract the timestamp and a key from a record.
+     *
+     * @since 10.1
+     */
+    default <M extends Externalizable> List<Latency> getLatencyPerPartition(String name, String group,
+            Function<M, Long> timestampExtractor, Function<M, String> keyExtractor) {
```

Comment:
Don't you want to `@Deprecate` all the methods that don't take a codec to force callers to be explicit?